### PR TITLE
refactor: add Token.toString()

### PR DESCRIPTION
**Pull request by WalterBright**
Improves encapsulation, gets rid of some strlen()'s

### CODE REVIEW

File: src/cond.d [hunk 1]
```diff
@@ -30,13 +30,8 @@ private __gshared Identifier idAssert;
...
 {
-    const(char)* s;
-
-    s = Token.toChars(TOKunittest);
-    idUnitTest = Identifier.idPool(s, strlen(s));
-
-    s = Token.toChars(TOKassert);
-    idAssert   = Identifier.idPool(s, strlen(s));
+    idUnitTest = Identifier.idPool(Token.toString(TOKunittest));
+    idAssert   = Identifier.idPool(Token.toString(TOKassert));
```
Comment (yebblies) [inline src/cond.d]: These should be moved to idgen.

File: src/tokens.d [hunk 2]
```diff
@@ -570,7 +570,7 @@ extern (C++) struct Token
         Identifier ident;
     }
 
-    static __gshared const(char)*[TOKMAX] tochars =
+    extern (D) private __gshared const(char)[][TOKMAX] tochars =
```
Comment (yebblies) [inline src/tokens.d]: Should be `string[TOKMAX]`

**Comment by UplinkCoder** [comment]
I like this change. :+1: 
It makes the codebase more D-ish.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**